### PR TITLE
Inv 53 js to ts

### PR DESCRIPTION
**Pull request by nskoretz**
**Changed:**
- Replaced VueResource with Axios for our API handing
- Fixed indentation in .js and files
- Failed /login call tries to return HTTP 401
- Changed storage of login variables to the Vuex Store instead of the
component.
- Login.vue; Migrate login variables from component data to Vuex Store.
- Login.vue; Follow the 'vue-class-component' layout for Typescript Vue
components. This adds a lot more inherent type checks.
- Login.vue; attemptLogin method now commits login information to the
store along with dispatching the 'postLogin' Store action.
- Signup.vue now uses axios for requests.
- Fix Login.vue err styling. 

**Added**

- Login.vue; Add a <div> that lists field requirements that is
revealed when the user types.
- Login.vue; Add CSS that aligns text to the left.
- interfaces/; Add a location for defining Typescript Interfaces.
- interfaces/LinkAction.ts; Interface for LinkAction data that is passed
as props to Navbar component.

**Removed**
- Login.vue; Remove computed methods that 'get' and 'set' the store.
- /server/public/js*; All files in this directory
- /server/src/mixins*; All files in this directory
- Remove unneeded comments in Login.vue and main.ts

### CODE REVIEW

File: server/src/store.ts [hunk 1]
```diff
@@ -1,16 +1,57 @@
 import Vue from 'vue';
 import Vuex from 'vuex';
+import axios from 'axios';
+// import VueResource from 'vue-resource';
```
Comment (HugoKlepsch) [inline server/src/store.ts]: Forgot to remove this?

File: server/src/store.ts [hunk 2]
```diff
@@ -1,16 +1,57 @@
...
+      response: {},
+    },
+  }, // }}}
 
-  },
-  mutations: {
+  // Mutations commit and track state changes.
+  mutations: { // {{{
+    setLoginPassword( state, password ) {
+      state.login.password = password;
```
Comment (HugoKlepsch) [inline server/src/store.ts:20]: Although this pattern is pretty verbose, lets keep going with it.

File: server/src/store.ts [hunk 3]
```diff
@@ -1,16 +1,57 @@
...
+Vue.use( Vuex );
 
 export default new Vuex.Store({
-  state: {
+  // Same as 'data' for a Vue component, except accessable by all components.
+  state: { // {{{
+    login: {
+      username: '',
+      password: '',
+      response: {},
```
Comment (HugoKlepsch) [inline server/src/store.ts:13]: Does all state that we want to save need to go in here?

File: server/src/views/Login.vue [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
           <img src='/img/face-dark-empty.png'>
         </div>
-        <div v-bind:class='{invalidForm: loginError}' class='form'>
+        <div v-if='isReqsRevealed' id='reqs'>
+          <ul>
+            <li><em>Username must be between 4 & 30 alphanumeric characters</em></li>
+            <li><em>Password must be at least 4 characters</em></li>
+          </ul>
+        </div>
+        <!-- <div class='form'> -->
```
Comment (HugoKlepsch) [inline server/src/views/Login.vue:15]: ?

File: server/src/views/Login.vue [hunk 5]
```diff
@@ -25,15 +38,16 @@
...
             />
 
           <button
             type='submit'
             v-on:click="attemptLogin"
             >Login
           </button>
-        </div>
+        </form>
+        <!-- </div> -->
```
Comment (HugoKlepsch) [inline server/src/views/Login.vue:50]: same

**Review by HugoKlepsch — COMMENTED** [review]
Nitpicky stuff

File: server/src/store.ts (hunk 1 above)
Comment (nskoretz) [reply server/src/store.ts]: Yes

File: server/src/store.ts (hunk 2 above)
Comment (nskoretz) [reply server/src/store.ts:20]: This is the pattern to mutate data in the store. Not much we can do about it. We can opt not to put things in the store if you want.

File: server/src/store.ts (hunk 3 above)
Comment (nskoretz) [reply server/src/store.ts:13]: Yes, if you want it to persist. But we can break the store into multiple files.

File: server/src/views/Login.vue (hunk 4 above)
Comment (nskoretz) [reply server/src/views/Login.vue:15]: posterity?

File: server/src/views/Login.vue (hunk 5 above)
Comment (nskoretz) [reply server/src/views/Login.vue:50]: same

**Review by HugoKlepsch — APPROVED** [review]
LGTM :+1:

**Pull request merged**